### PR TITLE
Tokemak deposit

### PR DESCRIPTION
**Pull request by to-sh-1**
Note: I haven't added in the abstraction for depositing, I have created a ticket to handle that later here https://github.com/shapeshift/foxy/issues/17

- Got rid of the mock FOX and used a mainnet fork to interact with TOKE directly
- Removed the test deploys since we aren't using mock FOX anymore
- Added mainnet env variable and use that when instantiating a local node
- Added TOKE deposit to the stake function
- Included the FOX balance in TOKE to the contractSupply to give correct rewards

### CODE REVIEW

File: deploy/FoxStaking.ts
Comment (0xean) [inline deploy/FoxStaking.ts]: I would not hardcode the fox address.  Plan is that in the future other DAOs may want to use this same set of contracts so allowing the token address to be easily passed in should help facilitate this and the gas savings is trivial with immutable vs constant

File: hardhat.config.ts
Comment (0xean) [inline hardhat.config.ts:33]: No arguments regarding the forking from mainnet strategy to test, but we still may want to consider a testnet deployment in the coming weeks and how we would interact with that.   Will leave that up to you, but we may need a mock fox token and a mock reactor in that case.

File: src/contracts/FoxStaking.sol
Comment (0xean) [inline src/contracts/FoxStaking.sol]: see previous  comment re: hardcoding FOX

Comment (0xean) [reply src/contracts/FoxStaking.sol]: would even recommend that we actually use a variable name different than FOX in the case we do deploy this for other DAOs in the future.

File: src/contracts/FoxStaking.sol
Comment (0xean) [inline src/contracts/FoxStaking.sol]: If you want unlimited approval please use MAXUINT256

File: test/fox-abi.ts
Comment (0xean) [inline test/fox-abi.ts]: We shouldn't need this as we have the generic ERC20 abi.

File: test/foxStakingTest.ts
Comment (0xean) [inline test/foxStakingTest.ts]: This seems a bit fickle to rely on an on-chain address not changing for our tests to work.

Given that FOX is a generic ERC20, I think it would be better to just deploy a fake ERC20 that we entirely control for the durability of the tests. 

This also helps because we want this to work with any ERC20 in the future.

**Review by 0xean — CHANGES_REQUESTED** [review]
Nice work @toshiSat.  I left a bunch of comments, happy to chat through them if you have questions.

File: hardhat.config.ts
Comment (to-sh-1) [reply hardhat.config.ts:33]: I think this is a good structure if we do use that, but I am going to leave it for now.  It was spawning multiple versions of the staking contract so to keep things simple I want to only use what we are currently using.  If we do decide to mock the reactor I will re-add this pattern

File: test/foxStakingTest.ts
Comment (to-sh-1) [reply test/foxStakingTest.ts]: Solved by setting block number

File: deploy/FoxStaking.ts
Comment (to-sh-1) [reply deploy/FoxStaking.ts]: If there is time, I would love to try the same proxy contract pattern as TOKE.  I think having this in mind it would be a good way to do it

File: src/contracts/FoxStaking.sol
Comment (to-sh-1) [reply src/contracts/FoxStaking.sol]: For the renaming, I think that should be its own PR.  We have FOX labelled everywhere.  We should discuss this and decide on a better name

Comment (to-sh-1) [reply src/contracts/FoxStaking.sol]: https://github.com/shapeshift/foxy/issues/19

**Comment by to-sh-1** [comment]
On top of the review changes, I added the tokePool address to the deploy script

File: test/foxStakingTest.ts
Comment (0xean) [inline test/foxStakingTest.ts]: something like `import ERC20 from "@openzeppelin/contracts/build/contracts/ERC20.json"; ` and `ERC20.abi` should work and be a little cleaner.

Comment (0xean) [reply test/foxStakingTest.ts]: Also not sure the `Foxy__factory` is being used.

**Review by 0xean — APPROVED** [review]
small comments but 👍  lgtm

**Pull request merged**
